### DISCUSSION TITLE
Xray Core vs Sing-box

### DISCUSSION THREAD

**Discussion by bart3nder** [XTLS/Xray-core#2023] [Q&A]
Hello y'all!
As the title says, I'm curious about what is the diffrence between Xray Core and Sing-box, in terms of stability, security, performance, speed and specially features. I saw some comparisons that show sing-box uses less memory and battery, but I wanted to know your opinions too.

**Comment by RPRX** [comment]
> I saw some comparisons that show sing-box uses less memory and battery, but I wanted to know your opinions too.

You mean [sing-box vs v2fly-core](https://sing-box.sagernet.org/features/) right?

Well, Xray-core has [better performance](https://github.com/XTLS/Xray-core/releases/tag/v1.0.0) and [optimized memory usage by default](https://github.com/XTLS/Xray-core/releases/tag/v1.1.4) than v2fly-core, from the very beginning.

But people always forget them. Perhaps we should list our significant accomplishments in `README.md`, as many other projects do.

**Comment by bart3nder** [comment]
That would be great, Thanks 👍

**Comment by MahdiAkrami01** [comment]
is still sing-box performance bit the xray-core in 2025 ?
